From: Lucas Kanashiro <lucas.kanashiro@canonical.com>
Date: Thu, 20 Aug 2020 11:12:52 -0300
Subject: Backport update-smart-drivedb.8.in with GPG verification support

The paths (most of things were pointing users to /usr/local and not the
properly path where files are installed via Debian package), and the GPG
public key ID were updated on top of the upstream version.

Origin: upstream, https://github.com/smartmontools/smartmontools/blob/RELEASE_7_1/smartmontools/update-smart-drivedb.8.in
Bug-Ubuntu: #1874362
Last-Updated: 2020-08-20
---
 update-smart-drivedb.8.in | 133 +++++++++++++++++++++++++++++-----------------
 1 file changed, 85 insertions(+), 48 deletions(-)

diff --git a/update-smart-drivedb.8.in b/update-smart-drivedb.8.in
index 1517be5..df2b634 100644
--- a/update-smart-drivedb.8.in
+++ b/update-smart-drivedb.8.in
@@ -1,27 +1,38 @@
 .ig
 Copyright (C) 2013 Hannes von Haugwitz <hannes@vonhaugwitz.com>
-Copyright (C) 2014-16 Christian Franke
+Copyright (C) 2014-19 Christian Franke
 
-$Id: update-smart-drivedb.8.in 4223 2016-02-26 20:18:40Z chrfranke $
+SPDX-License-Identifier: GPL-2.0-or-later
 
-This program is free software; you can redistribute it and/or modify
-it under the terms of the GNU General Public License as published by
-the Free Software Foundation; either version 2, or (at your option)
-any later version.
-
-You should have received a copy of the GNU General Public License
-(for example COPYING); If not, see <http://www.gnu.org/licenses/>.
+$Id$
 
 ..
+.\" Macros borrowed from pages generated with Pod::Man
+.de Sp \" Vertical space (when we can't use .PP)
+.if t .sp 0.4v
+.if n .sp
+..
+.de Vb \" Begin verbatim text
+.ft CW
+.nf
+.ne \\$1
+..
+.de Ve \" End verbatim text
+.ft R
+.fi
+..
+.\" Use groff extension \(aq (apostrophe quote, ASCII 0x27) if possible
+.ie \n(.g .ds Aq \(aq
+.el       .ds Aq '
 .TH UPDATE-SMART-DRIVEDB 8 "CURRENT_SVN_DATE" "CURRENT_SVN_VERSION" "SMART Monitoring Tools"
 .SH NAME
 update-smart-drivedb \- update smartmontools drive database
-
+.Sp
 .SH "SYNOPSIS"
 .B update-smart-drivedb
 .RI [ OPTIONS ]
 .RI [ DESTFILE ]
-
+.Sp
 .SH "DESCRIPTION"
 .\" %IF NOT OS ALL
 .\"! [This man page is generated for the OS_MAN_FILTER version of smartmontools.
@@ -30,13 +41,12 @@ update-smart-drivedb \- update smartmontools drive database
 .\" %ENDIF NOT OS ALL
 .B update-smart-drivedb
 updates
-.B /usr/local/share/smartmontools/drivedb.h
+.B /var/lib/smartmontools/drivedb/drivedb.h
 or
 .I DESTFILE
-from smartmontools SVN repository.
-
-It tries to download first from the current branch and then from
-trunk. The tools used for downloading are either
+from branches/RELEASE_6_0_DRIVEDB of smartmontools SVN repository.
+.PP
+The tools used for downloading are either
 .BR curl (1),
 .BR wget (1),
 .BR lynx (1),
@@ -50,20 +60,23 @@ trunk. The tools used for downloading are either
 .\" %ENDIF OS OpenBSD
 or
 .BR svn (1).
-
+.PP
+The downloaded file is verified with OpenPGP/GPG key ID DFD22559.
+The public key block is included in the script.
+.PP
 The old file is kept if the downloaded file is identical (ignoring
 the differences in Id string) otherwise it is moved to
 .BR drivedb.h.old .
-
+.Sp
 .SH "OPTIONS"
 .TP
 .B \-s SMARTCTL
 Use the
 .BR smartctl (8)
 executable at path SMARTCTL for drive database syntax check.
-The form \'\-s \-\' disables the syntax check.
+The form \*(Aq\-s \-\*(Aq disables the syntax check.
 The default is
-.BR /usr/local/sbin/smartctl .
+.BR /usr/sbin/smartctl .
 .TP
 .B \-t TOOL
 Use TOOL for download.
@@ -79,23 +92,32 @@ TOOL is one of:
 The default is the first one found in PATH.
 .TP
 .B \-u LOCATION
-Use URL of LOCATION for download. LOCATION is one of:
+Use URL of LOCATION for download.
+LOCATION is one of:
+.br
+.I github
+(GitHub mirror of SVN repository),
 .br
 .I sf
-(Sourceforge code browser via HTTPS),
+(Sourceforge code browser),
 .br
 .I svn
-(SVN repository via HTTPS),
+(SVN repository),
 .br
 .I svni
-(SVN repository via HTTP),
+(SVN repository via HTTP instead of HTTPS),
 .br
 .I trac
-(Trac code browser via HTTPS).
+(Trac code browser).
 .br
 The default is
 .IR svn .
 .TP
+.B \-\-trunk
+Download from SVN trunk.
+This requires \*(Aq\-\-no\-verify\*(Aq unless the trunk version is still
+identical to branches/RELEASE_6_0_DRIVEDB.
+.TP
 .B \-\-cacert FILE
 Use CA certificates from FILE to verify the peer.
 .TP
@@ -104,63 +126,78 @@ Use CA certificate files from DIR to verify the peer.
 .TP
 .B \-\-insecure
 Don't abort download if certificate verification fails.
-This option is also required if a HTTP URL is selected with \'-u\' option.
+This option is also required if a HTTP URL is selected with \*(Aq\-u\*(Aq
+option.
+.TP
+.B \-\-no\-verify
+Don't verify signature with GnuPG.
+.TP
+.B \-\-export\-key
+Print the OpenPGP/GPG public key block.
 .TP
 .B \-\-dryrun
 Print download commands only.
 .TP
 .B \-v
 Verbose output.
-
+.Sp
 .SH "EXAMPLES"
-.nf
+.Vb 2
 # update-smart-drivedb
-/usr/local/share/smartmontools/drivedb.h updated from branches/RELEASE_6_0_DRIVEDB
-.fi
-
+/var/lib/smartmontools/drivedb/drivedb.h updated from \e
+branches/RELEASE_6_0_DRIVEDB
+.Ve
+.Sp
 .SH "EXIT STATUS"
 The exit status is 0 if the database has been successfully
-updated. If an error occurs the exit status is 1.
-
+updated.
+If an error occurs the exit status is 1.
+.Sp
 .SH FILES
 .TP
-.B /usr/local/sbin/update-smart-drivedb
+.B /usr/sbin/update-smart-drivedb
 full path of this script.
 .TP
-.B /usr/local/sbin/smartctl
+.B /usr/sbin/smartctl
 used to check syntax of new drive database.
 .TP
-.B /usr/local/share/smartmontools/drivedb.h
+.B /var/lib/smartmontools/drivedb/drivedb.h
 current drive database.
 .TP
-.B /usr/local/share/smartmontools/drivedb.h.old
-previous drive database.
+.B /var/lib/smartmontools/drivedb/drivedb.h.raw
+current drive database with unexpanded SVN Id string.
 .TP
-.B /usr/local/share/smartmontools/drivedb.h.error
-new drive database if rejected due to syntax errors.
+.B /var/lib/smartmontools/drivedb/drivedb.h.raw.asc
+signature file.
 .TP
-.B /usr/local/share/smartmontools/drivedb.h.lastcheck
+.B /var/lib/smartmontools/drivedb/drivedb.h.*old*
+previous files.
+.TP
+.B /var/lib/smartmontools/drivedb/drivedb.h.*error*
+new files if rejected due to errors.
+.TP
+.B /var/lib/smartmontools/drivedb/drivedb.h.lastcheck
 empty file created if downloaded file was identical.
-
+.Sp
 .SH AUTHORS
 \fBChristian Franke\fP.
 .br
 This manual page was originally written by
 .BR "Hannes von Haugwitz <hannes@vonhaugwitz.com>" .
-
+.Sp
 .SH REPORTING BUGS
 To submit a bug report, create a ticket in smartmontools wiki:
 .br
-<\fBhttp://www.smartmontools.org/\fP>.
+<\fBhttps://www.smartmontools.org/\fP>.
 .br
 Alternatively send the info to the smartmontools support mailing list:
 .br
-<\fBhttps://lists.sourceforge.net/lists/listinfo/smartmontools-support\fB>.
-
+<\fBhttps://listi.jpberlin.de/mailman/listinfo/smartmontools-support\fB>.
+.Sp
 .SH SEE ALSO
 \fBsmartctl\fP(8), \fBsmartd\fP(8).
-
+.Sp
 .SH PACKAGE VERSION
 CURRENT_SVN_VERSION CURRENT_SVN_DATE CURRENT_SVN_REV
 .br
-$Id: update-smart-drivedb.8.in 4223 2016-02-26 20:18:40Z chrfranke $
+$Id$
